Gopkg.toml

```diff
@@ -0,0 +1,63 @@
+# Gopkg.toml example
+#
+# Refer to https://golang.github.io/dep/docs/Gopkg.toml.html
+# for detailed Gopkg.toml documentation.
+#
+# required = ["github.com/user/thing/cmd/thing"]
+# ignored = ["github.com/user/project/pkgX", "bitbucket.org/user/project/pkgA/pkgY"]
+#
+# [[constraint]]
+#   name = "github.com/user/project"
+#   version = "1.0.0"
+#
+# [[constraint]]
+#   name = "github.com/user/project2"
+#   branch = "dev"
+#   source = "github.com/myfork/project2"
+#
+# [[override]]
+#   name = "github.com/x/y"
+#   version = "2.4.0"
+#
+# [prune]
+#   non-go = false
+#   go-tests = true
+#   unused-packages = true
+
+required = ["github.com/json-iterator/go"]
```

Comment:
GO dependency n00b question: Do we need this required statement if we then have the same repo specified as a contraint at line 42?

Reply:
Don't blame you on that. There are over 15 go dependency management systems ([https://github.com/golang/go/wiki/PackageManagementTools](https://github.com/golang/go/wiki/PackageManagementTools)) This one, `dep`, requires you to specify a package in the "required" array if you would like to impose a constraint on a package that is not directly (but indirectly) required by your application. In this case, the constraint solver was picking a bad version for this package (couldn't compile a dependency code branch). So I had to manually override.